exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/Metadata.java

```diff
@@ -1668,9 +1666,16 @@ public ColumnMetadata_v3(String[] name, PrimitiveTypeName primitiveType, Object
       return nulls;
     }
 
+    /**
+     * Checks that the column chunk has single value.
+     * Returns true if minValue and maxValue are null and
+     * column statistics was available.
+     *
+     * @return true if column has single value
+     */
     @Override
     public boolean hasSingleValue() {
-      return (minValue !=null && maxValue != null && minValue.equals(maxValue));
+      return minValue != null ? minValue.equals(maxValue) : (maxValue == null && isMetadataSet());
```

Comment:
1. It is clear that we should just compare minValue and maxValue.
2. In this case we should check that nulls == 0. Fixed that. Now for this case function returns false.
3. We should additionally check that nulls > 0. Fixed that.